### ISSUE TITLE
Changing SnapDropZone HighlightObject at runtime

### ISSUE THREAD

**Issue by TobySch** [thestonefox/VRTK#1520]
Hi there, 
I have an issue, more a Question, whether it is possible to change the HighlightObjectPrefab at runtime and reinitialize it on a at-run-time-initialized SnapDropZone object.

**What I want to do**
For explanation: I'm loading a 3D-Machine-Model at runtime. Afterwards I want to Initialize some DropZonePrefabs on some of the machineparts, so that they can be detached and attached again. To show they'r initial position I need to create a HighlightObject of the machinepart when the DropZone gets initialized. 

**What I have**
I've started by creating a Prefab, containing a "MachinePartWrapper" (which will keep the original part in it) and a DropZone-Object. I've referenced the PartWrapper as a HighlightObjectPrefab but while it gets initialized the wrapper is empty.
Here is how my hierarchy looks like:
+ GeneralWrapper
-+ PartWrapper (the part mesh will become a child of this object when the wrapper gets initialized)
-+ DropZone (the HighlightObjectPrefab is referenced by the PartWrapper)

**The issue**
When I try to reinitialize the HighlightObject, the HighlightContainer will be deleted or empty. If it is empty, there will be no MeshFilter and MeshRenderer components attachted to it. 
I've found out, that all works fine, when I use a DropZone, which is not added to the scene at runtime. I guess it is caused by the routines for detecting trigger collisions.

**What I've tried**
In fact that overwriting the HighlightObjectPrefab via script at runtime will cause some LogErrors, I've tried to deactivate the whole DropZone GameObject while setting the new properties and call the "InitaliseHighlightObject" method to create a new HighlightContainer. But in any case, there are no Mesh components getting created, or the HighlightContainer will disappear. (This works great on a not at runtime initialized dropzone)

**In Conclusion**
When I'm trying to change the HighlighObjectPrefab at runtime on a at-runtime-initialized DropZone the HighlightContainer will not be regenerated properly. Even when disabling the DropZone.

**How to reproduce**
- Drag a prefab containing a DropZone while runtime into your hierarchy
- Change the HighlightObjectPrfab property (On a DropZone which is placed in the scene by default the HighlightModel will be updated immediately)
- see what happens

Thank you for your support!

**Comment by TobySch** [comment]
I've found some kind of workaround. Possibly a not very elegant but simple solution. 

I have now created a extra prefab for the dropZone, which I'm instantiating while I'm creating the Wrapper. 
So my wrapper now loads the DropZone Prefab from the resources and sets the properties on its own instance. When he is finished I'm instantiating this instance as a GameObject into my scene. 
This works for me, but a method to change the HighlightObject at runtime would be quite handy.

Maybe I've helped someone with the same problem.

**Comment by TobySch** [comment]
I've figured out, that my workaround only works in Editor. If I build my application the HighlighObject wont be Initialized again. The initialization will be executed only the first time. 
Please help!

**Comment by thestonefox** [comment]
Snap Drop Zones are set up in a way that makes this fundamentally difficult to do. It actually clones the game object that is to be the highlighting object.

**Comment by TobySch** [comment]
I don't think, that the cloning causes this issue. The first time I initialize the DropZone at runtime it all works fine. But with the next ones the HighlightObject will be identical with the first one. So it seems that the cloning works fine, only the second (and following) initialization as HighlighObject causing some trouble.
I have also created a method to clone a mesh and bake it with all its childmeshes together. This operation works fine at runtime.
